backends/s3/credentials.go

```diff
@@ -0,0 +1,53 @@
+package s3
+
+import (
+	"os"
+
+	"github.com/minio/minio-go/v7/pkg/credentials"
+)
+
+// K8sSecretProvider is an implementation of Minio's credentials.Provider,
+// allowing to read credentials from Kubernetes secrets, as described in
+// https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure.
+type K8sSecretProvider struct {
```

Comment:
There is nothing really Kubernetes specific too this. Docker also has file based secrets. Perhaps call this `FileCredentialsProvider`?